src/mcp_solver.jl

```diff
@@ -102,6 +103,7 @@ function solve(solver::MCPSolver, problem, x0, params::AbstractVector{<:Abstract
         primals = variables[1:(problem.n)],
         equality_duals = variables[((problem.n + 1):(problem.n + problem.num_equality))],
         inequality_duals = variables[(problem.n + problem.num_equality + 1):end],
+        variables,
```

Comment:
If we add this to the output, it would be important to also add it to the backward pass. Otherwise, a user may use the result from the `variables` fields and get a wrong gradient.